File: schemas/crm/convex/custom_field_values.ts
// custom_field_values: stored values for custom fields on specific entity instances.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const customFieldValues = defineTable({
  customFieldId: v.id("custom_fields"),
  entityId: v.string(),
  value: v.string(),
  updatedAt: v.number(),
})
  .index("by_custom_field_id_and_entity_id", ["customFieldId", "entityId"])
  .index("by_entity_id", ["entityId"]);

File: schemas/crm/spacetimedb/custom_field_values.rs
// custom_field_values: Stored values for custom fields on specific entities.
// See README.md for full design rationale.
// Composite unique: (custom_field_id, entity_id) — enforce in reducer logic

use spacetimedb::Timestamp;

#[spacetimedb::table(name = custom_field_values, public)]
pub struct CustomFieldValue {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub custom_field_id: String, // UUID — FK → custom_fields.id (cascade delete)
    #[index(btree)]
    pub entity_id: String, // UUID — polymorphic reference to contact/company/deal/lead
    pub value: String,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
